docs/metrics.md

```diff
@@ -0,0 +1,435 @@
+
+The **method** is `click`.
+
+Example event registration:
+
+```javascript
+Services.telemetry.registerEvents("item_view.interaction", {
+  "item_view.click": {
+    methods: ["click"],
+    objects: ["edit_entry_button",
+		"delete_entry_button",
+        "show_password_button",
+        "copy_password_button",
+        "copy_username_button"],
+    extra: {"fxa_uid": uid}
+  }
+});
+```
+Example of recording a click to show password event:
+
+```javascript
+Services.telemetry.recordEvent("item_view.interaction",
+                    "click",
+                    "show_password_button",
+                    null,
+                    {"fxa_uid": uid});
+```
+
+### Events Related to Interactions in the Item Editors
+These events record actions users take in the item editors. There are separate views for editing existing items and for adding new entries, though the fields that exist in each are similar. Thus we have two **categories** of events: `new_item.interaction` and `edit_item.interaction`.
+
+For events related to the submission of actual item information, see the next section. The item editors have 5 fields: `title_field`, `origin_field` (URL for the credential), `username_field`, `password_field`, and `notes_field`. There are also three buttons: `save_entry_button`, `cancel_button`, and `toggle_password_button`. The latter toggles the visibility of characters in the password field.  
+
+
+The **methods** are `click` and `focus`.
+
+Example event registration (Substitute `new_item` with `edit_item` to register events for the existing item editor):
+
+```javascript
+Services.telemetry.registerEvents("new_item.interaction", {
+  "item_editor.click": {
+    methods: ["click", "focus"],
+    objects: ["title_field",
+        "origin_field",
+        "username_field",
+        "password_field",
+        "notes_field",
+        "save_entry_button",
+        "cancel_button",
+        "toggle_password_button"],
+    extra: {"fxa_uid": uid}
+  }
+});
+```
+
+Example of recording a click on the `save_entry_button`:
+
+```javascript
+Services.telemetry.recordEvent("item_editor.interaction",
+                    "click",
+                    "save_entry_button",
+                    null,
+                    {"fxa_uid": uid});
+```
+
+### Events Related to Submitting An Item Change from the Item Editors
+Events of **category** `item_change_submitted` record the actual submission of new item information to the datastore (either completely new entries or updates to existing items).
+
+
+In theory, these should be tied to equivalent events in the datastore, so that (for example) an `adding` event which is logged when a user submits a new item has a corresponding `added` event that gets logged when the item is successfully added to the datastore. This may need to change, but for simplicity's sake I've made it as close to the datastore events as possible (see next section). These events will be recorded as `item_change_submitted`.
```

Comment:
Consideration for the future:

* When autofill is implemented, we could still record an "item_change_submitted" event, adding a field to help disambiguate from the same activity in the editor; or a new event could be registered/recorded.